src/Nethermind/Nethermind.Blockchain/BlockTree.cs

```diff
@@ -868,19 +867,26 @@ public AddBlockResult SuggestBlock(Block block, BlockTreeSuggestOptions options
                 {
                     // TODO: this is here because storing block data is not transactional
                     // TODO: would be great to remove it, he?
-                    if (_logger.IsTrace) _logger.Trace($"Entering missing block info in {nameof(FindHeader)} scope when head is {Head?.ToString(Block.Format.Short)}");
-                    if (calculateTotalDifficulty)
+                    if (_logger.IsWarn) _logger.Warn($"Entering missing block info in {nameof(FindHeader)} scope when head is {Head?.ToString(Block.Format.Short)}. BlockHeader {header.ToString(BlockHeader.Format.FullHashAndNumber)}, CreateLevelIfMissing: {createLevelIfMissing}. BestKnownBeaconNumber: {BestKnownBeaconNumber}, BestKnownNumber: {BestKnownNumber}");
+                    if (createLevelIfMissing == false || (BestKnownBeaconNumber > BestKnownNumber && header.Number > BestKnownNumber))
+                    {
+                        if (_logger.IsWarn) _logger.Warn($"Ignoring creation of the level in {nameof(FindHeader)} scope when head is {Head?.ToString(Block.Format.Short)}. BlockHeader {header.ToString(BlockHeader.Format.FullHashAndNumber)}, CreateLevelIfMissing: {createLevelIfMissing}. BestKnownBeaconNumber: {BestKnownBeaconNumber}, BestKnownNumber: {BestKnownNumber}");
```

Comment:
Those messages are very user unfriendly, can we describe them with more context so the user could have a chance of understanding them and/or not be frightened by them?